test/controllers/thesis_controller_test.rb

```diff
@@ -327,6 +327,112 @@ def attach_files_to_records(tr, th)
     assert_select 'table#thesisQueue tbody tr', count: 1
   end
 
+  # ~~~~~~~~~~~~~~~ publication status report ~~~~~~~~~~~~~~~~
+  test 'anonymous users cannot see publication status report' do
+    # Note that nobody is signed in.
+    get thesis_publication_statuses_path
+    assert_response :redirect
+  end
+
+  test 'basic users cannot see publication status report' do
+    sign_in users(:basic)
+    get thesis_publication_statuses_path
+    assert_redirected_to '/'
+    follow_redirect!
+    assert_select 'div.alert', text: 'Not authorized.', count: 1
+  end
+
+  test 'submitters cannot see publication status report' do
+    sign_in users(:transfer_submitter)
+    get thesis_publication_statuses_path
+    assert_redirected_to '/'
+    follow_redirect!
+    assert_select 'div.alert', text: 'Not authorized.', count: 1
+  end
+
+  test 'processors can see publication status report' do
+    sign_in users(:processor)
+    get thesis_publication_statuses_path
+    assert_response :success
+  end
+
+  test 'thesis_admins can see publication status report' do
+    sign_in users(:thesis_admin)
+    get thesis_publication_statuses_path
+    assert_response :success
+  end
+
+  test 'admins can see publication status report' do
+    sign_in users(:admin)
+    get thesis_publication_statuses_path
+    assert_response :success
+  end
+
+  test 'publication status report shows theses with any status by default' do
+    # Calculate counts based on current test db, since these numbers will change as the fixtures do
+    thesis_count = Thesis.all.count
+
+    sign_in users(:processor)
+    get thesis_publication_statuses_path
+    assert_select 'table#thesisQueue tbody tr', count: thesis_count
+  end
+
+  test 'publication status report allows filtering by term' do
+    # Calculate counts based on current test db, since these numbers will change as the fixtures do
+    thesis_count = Thesis.all.count
+    term_count = Thesis.where(grad_date: '2018-09-01').count
```

Comment:
Minor suggestion, but checking that the `term_count` is less than the `thesis_count` for this would ensure we aren't just checking that nothing happens when the filter are applied. I'm not sure it's necessary, but it might be useful to know it's a smaller list we are getting at the end.

Reply:
That's a good callout. I just pushed up a commit that I think addresses this.